### ISSUE TITLE
Async initialization of a client

### ISSUE THREAD

**Issue by verdie-g** [aerospike/aerospike-client-csharp#133]
In some cases, a client is lazily initialized when we need, making the initialization in the hot path of a request. It would be great in that case no threads were blocked to avoid impacting the throughput of the thread pool.

Could we have an API like this one
```csharp
AsyncClient client = new(/* ... */);
await client.StartAsync();
```
or safer
```
AsyncClient client = await AsyncClient.CreateAsync(/* ... */)
```

**Comment by shannonklaus** [comment]
Yes, we can have an API like this in the async awaitable client. Thanks for bringing this to my attention.
